### DISCUSSION TITLE
Background Tasks in single connection

### DISCUSSION THREAD

**Discussion by nikita-b** [fastapi/fastapi#9525] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@v1_router.get("/namespace/{backend}/generate_certificates")
def create_nomad_namespace(backend: str, namespace: str, background_tasks: BackgroundTasks):
    """Create namespace in Nomad"""
    background_tasks.add_task(
        run_batch_job,
    )


def run_batch_job(backend_name: str, namespace: str, nomad_file: str) -> bool:
    """Run batch Job in nomad"""
    **something**
```

### Description

Hello,

I have a question regarding `BackgroundTasks`.

I have often used an endpoint with `BackgroundTasks`. Everything works perfectly if I send requests in different connections but if I use the same connection:
```
curl -k -X POST  http://127.0.0.1:8000/api/v1/debug/debug http://127.0.0.1:8000/api/v1/debug/debug
```
Background tasks are executed sequentially and the second task waits until the first one ends.

Is it work as expected? Can I fix it with `BackgroundTasks` or do I need to implement the custom solution?

### Operating System

Linux

### FastAPI Version

0.95.0

### Python Version

3.11

**Comment by Kludex** [comment, marked as answer]
This is a bug. Check Starlette issues.

**Reply by nikita-b** [reply]
Oh, thanks. I already thought that I'm crazy.

**Reply by Sooyoung98** [reply]
I am also waiting for this bug to be fixed :) This was really an unexpected behavior and I feel like I should use other multithreading packages :(

**Reply by amacfie-tc** [reply]
see https://github.com/tiangolo/fastapi/pull/9636
